wp-cache-export.php

```diff
@@ -0,0 +1,472 @@
+
+  static $MESSAGES = array();
+
+
+  /**
+   * Instantiate the class to enable the importing and exporting features for the WP Super Cache plugin.
+   *
+   * Caches the location of the WP Super Cache plugin configuration file and the sample configuration file.
+   * Adds two actions to the WP Super Cache admin page load that determine if an import or export
+   * should be preformed.
+   *
+   * Defines the static $MESSAGES array that displays the success or error messages in the upload form.
+   *
+   * @uses  $wp_cache_config_file The location of the configuration file.
+   * @uses  $wp_cache_config_file_sample The location of the sample configuration file.
+   *
+   * @since  1.4.4
+   */
+
+  function __construct() {
+    global $wp_cache_config_file, $wp_cache_config_file_sample;
+
+    self::$cache_config_file = $wp_cache_config_file;
+    self::$cache_config_file_backup = str_replace( '.php', '-backup.php', self::$cache_config_file );
+
+    self::$MESSAGES = array(
+      0 =>  array( 'success', __( 'Settings imported', 'wp-super-cache' ) ),
+      1 =>  array( 'warning', __( 'Please upload an exported WP Super Cache settings file.', 'wp-super-cache' ) ),
+      2 =>  array( 'error', __( 'Unable to import the uploaded JSON file. Please export the settings and import them again.', 'wp-super-cache' ) ),
+      3 =>  array( 'error', __( 'Unable to create backup settings file. Please check that the wp-content folder is writable via the <em>chmod</em> command on the server.', 'wp-super-cache' ) ),
+      4 =>  array( 'error', __( 'Unable to remove the backup file. Please check that the wp-content folder is writable via the <em>chmod</em> command on the server.', 'wp-super-cache' ) ),
+      5 =>  array( 'success', __( 'All settings have been restored.', 'wp-super-cache' ) ),
+      6 =>  array( 'success', __( 'All backup settings have been removed permanently.', 'wp-super-cache' ) ),
+    );
+
+    add_action( 'load-settings_page_wpsupercache', array( $this, 'export' ) );
+    add_action( 'load-settings_page_wpsupercache', array( $this, 'import' ) );
+    add_action( 'load-settings_page_wpsupercache', array( $this, 'restore' ) );
+    add_action( 'load-settings_page_wpsupercache', array( $this, 'remove' ) );
+  }
+
+  /**
+   *  The form HTML that shows in the Import/Export tab of the WP Super Cache plugin admin page.
+   *
+   * Provides a basic form for exporting the plugin's settings.
+   * Provides a basic upload field for importing settings.
+   *
+   * A success message is displayed if new settings have been imported correctly.
+   * An error message is displayed if the settings cannot be imported.
+   *
+   *  @since  1.4.4
+   */
+  static function form() {
+    ?>
+
+    <?php if (  isset( $_GET['message'] ) ):  ?>
+      <div id="message" class="notice notice-<?php echo self::$MESSAGES[ $_GET['message'] ][0] ; ?> is-dismissible">
+        <p><?php echo self::$MESSAGES[ $_GET['message'] ][1]  ?></p>
```

Comment:
Good call. I replaced the translated text with their escaped html counterparts 802e94c5b.